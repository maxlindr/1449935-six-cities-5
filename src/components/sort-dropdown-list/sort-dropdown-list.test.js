import React from 'react';
import renderer from 'react-test-renderer';
import SortDropdownList from './sort-dropdown-list';

const SortType = {
  POPULAR: `popular`,
  TO_HIGH: `low-to-high`,
  TO_LOW: `high-to-low`,
  TOP_RATED: `top-rated`
};

const OPTIONS = {
  [SortType.POPULAR]: `Popular`,
  [SortType.TO_HIGH]: `Price: low to high`,
  [SortType.TO_LOW]: `Price: high to low`,
  [SortType.TOP_RATED]: `Top rated first`,
};

describe(`SortDropdownList`, () => {
  it(`должен корректно отображаться при isOpened=true`, () => {
    const component = renderer.create(
        <SortDropdownList
          isOpened={true}
          options={OPTIONS}
          activeOption={SortType.POPULAR}
          onToggle={() => {}}
          onOptionClick={() => {}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`должен корректно отображаться при isOpened=false`, () => {
    const component = renderer.create(
        <SortDropdownList
          isOpened={false}
          options={OPTIONS}
          activeOption={SortType.POPULAR}
          onToggle={() => {}}
          onOptionClick={() => {}}
          onChange={() => {}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
