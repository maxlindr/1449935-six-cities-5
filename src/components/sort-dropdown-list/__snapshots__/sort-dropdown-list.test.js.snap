// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortDropdownList должен корректно отображаться при isOpened=false 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom"
  >
    <li
      className="places__option places__option--active"
      data-value="popular"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      data-value="low-to-high"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      data-value="high-to-low"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      data-value="top-rated"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`SortDropdownList должен корректно отображаться при isOpened=true 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option places__option--active"
      data-value="popular"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      data-value="low-to-high"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      data-value="high-to-low"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      data-value="top-rated"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
